pkg/scalers/postgresql_scaler.go

```diff
@@ -42,12 +41,46 @@ type postgreSQLScaler struct {
 }
 
 type postgreSQLMetadata struct {
-	targetQueryValue           float64
-	activationTargetQueryValue float64
-	connection                 string
-	query                      string
+	TargetQueryValue           float64 `keda:"name=targetQueryValue, order=triggerMetadata, optional"`
+	ActivationTargetQueryValue float64 `keda:"name=activationTargetQueryValue, order=triggerMetadata, optional"`
+	Connection                 string  `keda:"name=connection, order=authParams;resolvedEnv, optional"`
+	Query                      string  `keda:"name=query, order=triggerMetadata"`
 	triggerIndex               int
 	azureAuthContext           azureAuthContext
+
+	Host     string `keda:"name=host, order=authParams;triggerMetadata, optional"`
+	Port     string `keda:"name=port, order=authParams;triggerMetadata, optional"`
+	UserName string `keda:"name=userName, order=authParams;triggerMetadata, optional"`
+	DBName   string `keda:"name=dbName, order=authParams;triggerMetadata, optional"`
+	SslMode  string `keda:"name=sslmode, order=authParams;triggerMetadata, optional"`
+
+	Password string `keda:"name=password, order=authParams;resolvedEnv, optional"`
+}
+
+func (p *postgreSQLMetadata) Validate() error {
+	if p.Connection == "" {
```

Comment:
Maybe it's better to return directly if p.Connection != ""

Reply:
I agree, that might be a little bit nicer, although two nested `if` statements are imho still within reasonable boundaries of readability so I'm ok with either :)